File: java/src/pattern/strategy/duck/MallardDuck.java
package pattern.strategy.duck;

public class MallardDuck extends Duck
{
    public MallardDuck()
    {
	super.Duck();
	flyBehaviour = new FlyWithWings();
	quackBehaviour = new Quack();
    }

    public void display()
    {
	System.out.println("I'm a real Mallard duck!");
    }
}

File: php/src/pattern/strategy/duck/MallardDuck.php
<?php

namespace pattern\strategy\duck;

class MallardDuck extends Duck
{
    public function __construct()
    {
	parent::__construct();
	$this->_flyBehaviour = new FlyWithWings();
	$this->_quackBehaviour = new Quack();
    }

    public function display()
    {
	echo "I'm a real Mallard duck!\n";
    }
}
